src/Symfony/Component/Yaml/Parser.php

```diff
@@ -751,7 +751,8 @@ private function parseValue($value, $flags, $context)
                 $lines[] = trim($this->currentLine);
 
                 // quoted string values end with a line that is terminated with the quotation character
-                if ('' !== $this->currentLine && substr($this->currentLine, -1) === $quotation) {
+                $escapedLine = str_replace(['\\\\', '\\"'], '', $this->currentLine);
```

Comment:
This would still break when the strings ends with something like `bar\\\\\\"`, wouldn't it?

Reply:
Did you give the source string or a php string with escaped backslashes?
Anyway both cases look correct:
```php
$backslash = '\\';
$a = "bar$backslash$backslash$backslash$backslash$backslash$backslash\""; // becomes `bar"` after repalce
$b = "bar$backslash$backslash$backslash\""; // becomes `bar` after replace
```
6 backslashes and quote in the source string is 3 escaped backslashes, the quote after them closes a quoted string.
3 backslashes and quote in the source string is 1 escaped backslash and 1 escaped quote, string is not closed.

Reply:
never mind, I think I had some wrong thought here 😄